modules/plugins/mini/indentscope/config.nix

```
@@ -3,13 +3,37 @@
   lib,
   ...
 }: let
+  inherit (lib.generators) mkLuaInline;
   inherit (lib.modules) mkIf;
   inherit (lib.nvim.dag) entryAnywhere;
   inherit (lib.nvim.lua) toLuaObject;
 
   cfg = config.vim.mini.indentscope;
 in {
   vim = mkIf cfg.enable {
+    autocmds = [
+      {
+        callback =
+          mkLuaInline
+          ''
+            function()
+              local ignore_filetypes = {
+                "help",
+                "neo-tree",
+                "notify",
+                "NvimTree",
+                "TelescopePrompt",
+              }
+              if vim.tbl_contains(ignore_filetypes, vim.bo.filetype) then
+                vim.b.miniindentscope_disable = true
+              end
+            end
+          '';
```

```suggestion
        callback = mkLuaInline ''
            function()
              local ignore_filetypes = {
                "help",
                "neo-tree",
                "notify",
                "NvimTree",
                "TelescopePrompt",
              }
              if vim.tbl_contains(ignore_filetypes, vim.bo.filetype) then
                vim.b.miniindentscope_disable = true
              end
            end
          '';
```

This is a nixfmt artifact. Please disable nixfmt while working with nvf.

---

I also think the ignore filetypes should be an option.

---

> This is a nixfmt artifact. Please disable nixfmt while working with nvf.

This is from stylua, do you suggest I keep the list in a single line?

---

> I also think the ignore filetypes should be an option.

I can add an option, let me make a followup

---

> > This is a nixfmt artifact. Please disable nixfmt while working with nvf.
> 
> This is from stylua, do you suggest I keep the list in a single line?

The function call and the arg string being in different lines cannot be Stylua, it's a Nix formatting issue. Either you are manually placing the multiline string, or a formatter is doing it for you. I don't remember alejandra doing that, which is why I assumed it's nixfmt (because nixfmt has a fetish for making everything into new lines)

---

> > > This is a nixfmt artifact. Please disable nixfmt while working with nvf.
> > 
> > 
> > This is from stylua, do you suggest I keep the list in a single line?
> 
> The function call and the arg string being in different lines cannot be Stylua, it's a Nix formatting issue. Either you are manually placing the multiline string, or a formatter is doing it for you. I don't remember alejandra doing that, which is why I assumed it's nixfmt (because nixfmt has a fetish for making everything into new lines)

It is from alejandra

```
❯ alejandra .
Checking style in 544 files using 2 threads.


Congratulations! Your code complies with the Alejandra style.
```

---

Does it still move the string down if you inline the definition like

```nix
 callback = mkLuaInline ''
 	-- lua here
 '';
```

This is a stylistic nit that matters little, but I'm not quite happy with wasting a line like current formatting does. If Alejandra still messes it up, we can forget about it.

---

No, apparently both are valid. It probably has to do with me adding #lua and then removing it. `mkLuaInline ''` is valid as per Alejandra but 
```
mkLuaInline #lua
''
```

gets forcefully formatted as

```
mkLuaInline
#lua
'' 
```

I will shift, I am sorry for the confusion

---

>I also think the ignore filetypes should be an option

Added an option

> The function call and the arg string being in different lines

Fixed